### DISCUSSION TITLE
WiFi.hostByName resolves to 255.255.255.255 when should return error

### DISCUSSION THREAD

**Discussion by gnalbandian** [esp8266/Arduino#7945] [Q&A] [answered]
Hi.

I'm having issues with WiFi.hostByName.
The function is returning 255.255.255.255 when trying to resolve google.com.er, which doesn't exists.
It should return error.

The function used internally is dns_gethostbyname.

Can someone else makes some test to figure out is this is an actual issue or is something particularly of my own code.

Thanks!

**Comment by d-a-v** [comment, marked as answer]
`255.255.255.255` is `INADDR_NONE`.
In current git version, the resulting IP can be tested with `.isSet()` which tests if IP address is neither `0.0.0.0` or `255.255.255.255`.
The `bool()` operator uses `::isSet()` so you can do that:
```cpp
    if (!ip)
        Serial.println("address not found");
```

Unfortunately v2.7.4 (latest release) `::isSet()` does not check for `NONE`.
v3.0 of this core will be out soon, in the meantime you can use git master, or the "stage" version if you are using PlatformIO, or an [unofficial snapshot release](https://d-a-v.github.io/esp8266/Arduino) for the arduino IDE board manager.

**Reply by gnalbandian** [reply]
Thanks. I really appreciate your answer.
Regard!
